### PR TITLE
VR-3755 repository admin creation and refactoring

### PR DESCRIPTION
**Pull request by evgeniilezhnin**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java [hunk 1]
```diff
@@ -154,31 +154,19 @@ private void createWorkspaceRoleBinding(
...
       Role datasetRead = roleService.getRoleByName(ModelDBConstants.ROLE_DATASET_READ_ONLY, null);
       switch (workspaceType) {
         case ORGANIZATION:
-          Organization org = (Organization) roleService.getOrgById(workspaceId);
-          roleService.createRoleBinding(
-              datasetAdmin,
-              new CollaboratorUser(authService, org.getOwnerId()),
-              datasetId,
-              ModelDBServiceResourceTypes.DATASET);
           if (datasetVisibility.equals(DatasetVisibility.ORG_SCOPED_PUBLIC)) {
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java:161]: My mistake if this was missed earlier but let us have ORG_SCOPED_PUBLIC visibility for repo and then move this to roleService. createWorkspaceRoleBinding

**Review by ravishetye — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java (hunk 1 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java:161]: https://github.com/VertaAI/modeldb/pull/479

File: backend/src/main/java/ai/verta/modeldb/versioning/blob/container/DatasetContainer.java [hunk 2]
```diff
@@ -47,24 +47,29 @@ public void process(
...
-          blobHashes.add(blobHash);
-          for (Map.Entry<String, AutogenS3DatasetComponentBlob> component :
-              componentHashes.entrySet()) {
-            if (!blobHashes.contains(component.getKey())) {
-              blobHashes.add(component.getKey());
-              S3DatasetComponentBlobEntity s3DatasetComponentBlobEntity =
-                  new S3DatasetComponentBlobEntity(
-                      component.getKey(), blobHash, component.getValue().toProto().build());
-              session.saveOrUpdate(s3DatasetComponentBlobEntity);
+        AutogenS3DatasetBlob autogenS3DatasetBlob = AutogenS3DatasetBlob.fromProto(s3);
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/blob/container/DatasetContainer.java:50]: why was this required?

File: backend/src/main/java/ai/verta/modeldb/versioning/blob/container/DatasetContainer.java [hunk 3]
```diff
@@ -73,33 +78,39 @@ public void process(
...
-      blobHashes.add(blobHash);
-      for (Map.Entry<String, AutogenPathDatasetComponentBlob> component :
-          componentHashes.entrySet()) {
-        if (!blobHashes.contains(component.getKey())) {
-          blobHashes.add(component.getKey());
-          PathDatasetComponentBlobEntity pathDatasetComponentBlobEntity =
-              new PathDatasetComponentBlobEntity(
-                  component.getKey(), blobHash, component.getValue().toProto().build());
-          session.saveOrUpdate(pathDatasetComponentBlobEntity);
+    AutogenPathDatasetBlob autogenPathDatasetBlob = AutogenPathDatasetBlob.fromProto(path);
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/blob/container/DatasetContainer.java:90]: why was this required?

Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/versioning/blob/container/DatasetContainer.java:90]: There was a bug when I was testing this feature in case of creating a notebook with an empty path.

I can move it to a separate PR if needed

Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/versioning/blob/container/DatasetContainer.java:90]: fine for now , but please avoid this in future

**Review by ravishetye — APPROVED** [review]
